RFCs/RFC-0: Lightweight RFC process.md

```diff
@@ -0,0 +1,69 @@
+
+This document will refer to that process as the “Request For Comments (RFC) process”.
+
+## Solution
+The proposed solution to address the problem described above is to have an individual author or a group of authors submit a proposal to the community in order to gather feedback and achieve consensus. The RFC follows the same format as used by this proposal.
+
+Much inspiration for this proposal has been drawn from [Apache Geode's RFC process](https://cwiki.apache.org/confluence/display/GEODE/Lightweight+RFC+Process), which inturn draws from Phil Calçado’s [Structured RFC Process](https://philcalcado.com/2018/11/19/a_structured_rfc_process.html).
+
+All RFCs are submitted via PRs which will be merged when approved.
+
+### Collaboration
+Comments and feedback should be provided on the PR and focus on the technical content.
+
+Authors should address all comments written by the deadline. This doesn't mean every comment and suggestion must be accepted and incorporated, but they should be carefully read and responded to.
+
+Every RFC has a lifecycle. The life cycle has the following phases:
+* **Active**: The RFC has been drafted and is open for comments. When making the PR, the doc should start out in this state.
+* **Accepted**: All the comments have been addressed and the changes proposed on this RFC have been accepted. This should be the state while merging the PR.
+* **Dropped**: The changes proposed on this RFC aren't in effect anymore, the document is kept for historical purposes and there is a new RFC that’s more current.
+
+### Approval
+The proposal should be posted with a date by which the author would like to see the approval decision to be made. How much time is given to comment depends on the size and complexity of the proposed changes, but ideally should be from 2 days to a week.
+
+### When to write an RFC?
+There is always the option of going straight to a pull request. However, for larger changes, it might be wise to de-risk the risk of rejection of the pull request by first gathering input from the community.
+
+It’s encouraged to write an RFC for any major change. A major change might be:
+* Addition of any major new feature or subsystem
+* Changes that impact existing, or introduce new public APIs.
+* Changes that need to be coordinated across multiple repositories.
+
+### How to write an RFC?
+1. Copy the RFC template (in same folder as this document) and write your proposal! It's up to the author's discretion to decide which section in the template make sense for their proposal. It's recommended to cover the problem the proposal is solving, who it affects, how you’re proposing to solve it, and answers to frequently asked questions. Explicitly listing the goals will also make it easier to retrospect later on the proposal.
+2. Add your RFC to the riff/rfcs source directory, update to the next unique number. 
```

Comment:
How should someone determine the next free RFC number? Suppose there are several RFC PRs in flight and some (recently) closed.

Reply:
I don't think there should be closed PRs. We should merge all RFC in either `Accepted` OR `Dropped` state, so numbering should be straight forward.